### ISSUE TITLE
CAST(x AS INT) triggers parsing errors in Trino dialect.

### ISSUE THREAD

**Issue by rileymcdowell** [sqlfluff/sqlfluff#5581]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wrote some Trino SQL with a `CAST(x AS INT)` in the query and linted it with SQLFluff. This generated a parsing error.

Upon further investigation, I found `CAST(x AS INTEGER)` works, so this problem seems to be related to `INT` vs `INTEGER` type aliases in Trino.

### Expected Behaviour

I expect `CAST (x AS INT)` to either be allowed, or be linted into `CAST (x AS INTEGER)` which is equivalent and is correctly parsed and linted by SQLFluff.

### Observed Behaviour

Here's an example SQL file:

```sql
SELECT CAST(1 AS INT)
```

Here's my SQLFluff command:

```bash
sqlfluff lint --dialect trino ./query.sql
```

And here's the output:

```txt
== [query.sql] FAIL
L:   1 | P:  15 |  PRS | Line 1, Position 15: Found unparsable section: 'as int'
WARNING: Parsing errors found and dialect is set to 'trino'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Create a one-line SQL file called `query.sql` with the following contents:

```sql
SELECT CAST(1 as INT)
```

Then run the following command with the latest SQLFluff release:

```bash
sqlfluff lint --dialect trino ./query.sql
```

### Dialect

Trino

### Version

```bash
sqlfluff --version
sqlfluff, version 2.3.5
```

```bash
python --version
Python 3.9.16
```

### Configuration

No configuration file was used. This can be triggered with the default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
